themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,105 @@
+* **The relationship between resources:** The relationships between resources---such as which resources are dependent upon other resources---is another factor that feeds into your Pulumi project structure. A "shared infrastructure layer" is often best handled in a separate project of its own, but may pull in resources that might otherwise have been kept with an application in its own project. (In some respects, this is a "counter-balance" to the first bullet point.)
+* **Resource lifecycle:** Do resources have a shared lifecycle, meaning they are always created, updated, or deleted together? (Note this is _not_ the same as a dependency on another resource!) If so, then those resources should be placed in the same project.
+* **Change rate of resources:** Some resources may change very infrequently. Consider the example of an AWS VPC and associated objects. Once it is created, it may not change for extended periods of time, if ever. On the other hand, some resources may change frequently (in other words, these resources have a high change rate). In a single project, combining resources that have a high change rate with resources that have a low change rate may be suboptimal.
+
+Now that you have an idea of some of the factors that affect how users structure their Pulumi projects, what are the advantages and disadvantages of each approach?
+
+For single projects, the key advantage is _simplicity._ There's no denying that a single project is simpler and less complex than working with multiple projects. This advantage, though, is also the key disadvantage. A single project does not afford the flexibility to let different teams manage different aspects of the infrastructure, for example, nor does it offer the ability to split resources that don't have a shared lifecycle. You can't run `pulumi destroy` on only _some_ of the resources in a project.
+
+For multiple projects, the key advantage is _flexibility._ It's true that the added flexibility comes at the cost of some added complexity (a future post will explore one aspect of that added complexity in the form of StackReferences). However, you gain the ability to separate infrastructure resources according to team ownership/responsibility, relationship to other resources (i.e., grouping resources in a "shared infrastructure layer"), or any of the other dimensions mentioned above.
+
+Finally, there is the consideration of what happens in the event of human error, and limiting the so-called "blast radius." With a single project, _all_ the resources in that project are within the blast radius---all of them could potentially be affected in the event of human error. Using separate Pulumi projects can limit the blast radius. It's not something that should drive the decision all by itself, but it _is_ a factor to be considered.
```

Comment:
I think this should be raised earlier because it applies even when there's 1 person working in isolation - you still need to limit blast radius. In fact, it was the first principle I learned for structuring IaC projects many moons ago.

Reply:
@jkodroff Where would you place the discussion blast radius, if not here?

Reply:
Left comment above.

Reply:
Addressed in e08be4c